### PR TITLE
Add info and code about querying dictionaries

### PR DESCRIPTION
**Pull request by MongoCaleb**
### Jira

https://jira.mongodb.org/browse/DOCSP-19341

### Staged Changes

https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-19341/sdk/dotnet/model-data/data-types/dictionaries/#usage-example

### IMPORTANT
We never manually edit files in the /generated/ directory. Code changes happen in the /examples/ directory only.

### CODE REVIEW

File: source/sdk/dotnet/model-data/data-types/dictionaries.txt [hunk 1]
```diff
@@ -46,12 +46,8 @@ The following code shows examples of dictionary types:
 
 Usage Example
 -------------
-The following code shows how to create, write to, and read from Dictionaries.
-
-.. note::
-
-   You cannot use Linq to query against a Realm Dictionary. Instead, 
-   use a string query.
+The following code shows how to create, write to, and read from Dictionaries,
```
Comment (mongomoe) [inline source/sdk/dotnet/model-data/data-types/dictionaries.txt:49]:
```suggestion
The following code shows how to create, write to, and read from Dictionaries
```

**Review by cbush — APPROVED** [review]

File: examples/dotnet/Examples/DataTypesSectionExamples.cs [hunk 2]
```diff
@@ -96,24 +96,33 @@ public async Task WorkWithDictionaries()
             });
```
Comment (papafe) [inline examples/dotnet/Examples/DataTypesSectionExamples.cs:96]: It's unnecessary to specify the type when using `realm.Add`.

File: examples/dotnet/Examples/DataTypesSectionExamples.cs [hunk 3]
```diff
@@ -78,16 +78,16 @@ public async Task WorkWithDictionaries()
                 Id = ObjectId.GenerateNewId().ToString()
```
Comment (papafe) [inline examples/dotnet/Examples/DataTypesSectionExamples.cs:78]: The ID is set, but that's not in the schema that is shown in the document

**Comment by papafe** [comment]
The changes that have been added are not showing the correct thing. What we want to show is that dictionaries can be transformed into a queryable live collection. For example like in https://github.com/realm/realm-dotnet/blob/ef50e4ce7c2359e4dcdef5110e6aecafcdd2f7ce/Tests/Realm.Tests/Database/RealmDictionaryTests.cs#L1076-L1111

Practically one can have a live collection of the Values of the dictionary using `AsRealmQueryable()`. Something like

```csharp
myStoreInventory.Plants.AsRealmQueryable().Where( p => p.Name == "Geranium") //This is a live collection 

```

**Review by papafe — CHANGES_REQUESTED** [review]

**Comment by MongoCaleb** [comment]
Latest version hasn't been released due to Core issues, so the updated code won't compile until then.

**Comment by MongoCaleb** [comment]
This PR will fail unit tests until the latest version of the SDK is released.

**Pull request merged**